packages/l/libmpdclient/xmake.lua

```diff
@@ -0,0 +1,15 @@
+package("libmpdclient")
+    set_homepage("https://musicpd.org/libs/libmpdclient/")
+    set_description("A stable, documented, asynchronous API library for interfacing MPD in the C, C++ & Objective C languages.")
+    add_urls("https://musicpd.org/download/libmpdclient/2/libmpdclient-$(version).tar.xz")
+    add_versions("2.19", "158aad4c2278ab08e76a3f2b0166c99b39fae00ee17231bd225c5a36e977a189")
+    add_deps("meson", "ninja")
+
+    on_install(function (package)
```

Comment:
If this package only support special platforms, you can write all supported platform name. like this:

```
on_install("macosx", "linux", "windows", function (package))
```

Reply:
It looks like it supports Linux and Windows, so I'll work on adding windows support to the file.

Reply:
Hmm, actually looking at the CI it doesn't seem like the file breaking it.
According to https://github.com/xmake-io/xmake-repo/pull/401/checks?check_run_id=2515333377#step:4:161, it fails on [checking](https://github.com/xmake-io/xmake-repo/pull/401/files#diff-92d2eb290c6a5922dbd8a73e2f15d740bf742df49239c5555f9eb964cdd06fa3R14) that the `connection.h` header has the correct function.
Is there anything extra I need to do to make it work for Windows?

Reply:
Also, it looks like Meson is unsupported in MingW (Msys2), which is why those CIs are breaking.

Reply:
Hmm, this is what appears to be the issue on Windows according to the CI:
```
> checking for c snippet(has_cfuncs)
checkinfo: ...gramdir\core\sandbox\modules\import\core\tool\linker.lua:75: @programdir\modules\core\tools\link.lua:155: LINK : fatal error LNK1181: cannot open input file 'mpdclient.dll.lib'
```

Reply:
Oh hmm, it looks like libmpdclient supports being built Windows through MingW. 
Could that perhaps be the issue?

It feels like it might be, so for now I'll limit installation to Linux.

Reply:
Actually, I noticed that mingw has it's own platform in xmake, and it's not that meson is unsupported; it's just that the current version isn't.

So I'm going to check and see what versions of meson are available in xmake, and try backtracking versions until one works.

Reply:
> Hmm, this is what appears to be the issue on Windows according to the CI:
> 
> ```
> > checking for c snippet(has_cfuncs)
> checkinfo: ...gramdir\core\sandbox\modules\import\core\tool\linker.lua:75: @programdir\modules\core\tools\link.lua:155: LINK : fatal error LNK1181: cannot open input file 'mpdclient.dll.lib'
> ```

Maybe you need to add some macro definitions to the dynamic library to enable symbol export, otherwise xxx.dll.lib will not be generated.

Reply:
> Actually, I noticed that mingw has it's own platform in xmake, and it's not that meson is unsupported; it's just that the current version isn't.
> 
> So I'm going to check and see what versions of meson are available in xmake, and try backtracking versions until one works.

you need add `@msys` to enable mingw on msys2/windows.

https://github.com/xmake-io/xmake-repo/blob/69cafbc69d217785effcb5f133d82ba19345b021/packages/m/meson/xmake.lua#L14

But I have not tested it yet.

Reply:
> Oh hmm, it looks like libmpdclient supports being built on Windows through MingW.
> Could that perhaps be the issue?
> 
> It feels like it might be, so for now I'll limit installation to Linux.

you can set `on_install("mingw@msys")` to enable this package only on mingw/msys2.